src/app/locationForm/number/LocationNumber.jsx

```diff
@@ -33,17 +121,24 @@ const mapStateToProps = (state, ownProps) => {
   const locationData = selectLocationData(state, locationId);
 
   return {
-    value: locationData && 
-              locationData.Phones &&
-              locationData.Phones[0] && 
-              locationData.Phones[0].number ? 
-              locationData.Phones[0].number : null,
+    value : {
+      number: locationData && 
+                locationData.Phones &&
+                locationData.Phones[0] && 
+                locationData.Phones[0].number ? 
+                locationData.Phones[0].number : null,
+      extension: locationData && 
```

Comment:
Doesn't seem necessary to repeat this set of checks for both number and extension - I think this could just be something along the lines of:
```
const phone = locationData && locationData.Phones && locationData.Phones[0];
const phoneFields = phone ? { number: phone.number, extension: phone.extension } : {};
return {
  value: phoneFields,
  locationData,
};
```

(Honestly, even the filtering of fields is probably unnecessary. The whole thing could probably just be `value: locationData && locationData.Phones && locationData.Phones[0]`.)

Reply:
And of course, a lot of util libraries just have functions for getting a nested property like this with 1 line (e.g. lodash's `get`). I guess the closest thing underscore has is `_.property`, though...

Reply:
Fixed in 85bab537c37f37de9a01b31aead81da23c950c40 per your suggestion.